File: binarySearch.js
const arrays = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];

function binarySearch(arrays, item) {
    let start = 0;
    let end = arrays.length - 1;
    let middle;
    let position = -1;

    while (start <= end) {
        middle = Math.floor((start + end) / 2);

        if (arrays[middle] === item) {
            position = middle;
            return position;
        }

        if (item < arrays[middle]) {
            end = middle - 1;
        } else {
            start = middle + 1;
        }
    }

    return position; 
}

console.log(binarySearch(arrays, 3)); 

File: binarySearch.java
public class binarySearch {

    public static void main(String[] args) { 

        int[] arrays = new int[]{1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
        int item = 5;
        int start = 0;  
        int end = arrays.length - 1;
        
        while (start <= end) { 
            int middle = (start + end) / 2;

            if (arrays[middle] == item) { 
                System.out.println("Element found at index: " + middle);
                return;
            }

            if (arrays[middle] > item) { 
                end = middle - 1;
            } else { 
                start = middle + 1;
            }
        }

        System.out.println("Element not found in the array.");
    }       
}
